src/lib/utilities/update-query-parameters.ts

```diff
@@ -54,3 +54,49 @@ export const updateQueryParameters = async ({
 
   return value;
 };
+
+export type QueryParameter = {
+  parameter: string;
+  value?: string | number | boolean;
+};
+
+type UpdateMultipleQueryParams = {
+  parameters: QueryParameter[];
+  url: URL;
+  goto?: typeof navigateTo;
+  clearParameters?: string[];
+};
+
+export const updateMultipleQueryParameters = async ({
```

Comment:
Maybe we combine this and `updateQueryParameters` at some point? But didn't want to touch existing logic with this PR.